internal/services/realtimeengine/common.go

```diff
@@ -21,7 +22,21 @@ func EnsureLicense(jwtWrapper wrappers.JWTWrapper) error {
 	if jwtWrapper == nil {
 		return errors.New("JWT wrapper is not initialized, cannot ensure license")
 	}
-	return nil
+
+	assistAllowed, err := jwtWrapper.IsAllowedEngine(params.CheckmarxOneAssistType)
+	if err != nil {
+		return errors.Wrap(err, "failed to check CheckmarxOneAssistType engine allowance")
+	}
+
+	aiAllowed, err := jwtWrapper.IsAllowedEngine(params.AIProtectionType)
+	if err != nil {
+		return errors.Wrap(err, "failed to check AIProtectionType engine allowance")
+	}
+
+	if aiAllowed || assistAllowed {
+		return nil
+	}
+	return errors.Wrap(err, "user does not have \"Checkmarx One Assist\" or \"AI Protection\" license")
```

Comment:
the next option can prevent unnessecary extra http request

```suggestion
	assistAllowed, err := jwtWrapper.IsAllowedEngine(params.CheckmarxOneAssistType)
	if err != nil {
		return errors.Wrap(err, "failed to check CheckmarxOneAssistType engine allowance")
	}
	
	if assistAllowed {
	    return nil
	}

	aiAllowed, err := jwtWrapper.IsAllowedEngine(params.AIProtectionType)
	if err != nil {
		return errors.Wrap(err, "failed to check AIProtectionType engine allowance")
	}

	if aiAllowed {
		return nil
	}
	return errors.Wrap(err, "user does not have \"Checkmarx One Assist\" or \"AI Protection\" license")
```